src/tools/sssctl/sssctl_config.c

```diff
@@ -34,6 +34,37 @@
 
 
 #ifdef HAVE_LIBINI_CONFIG_V1_3
+
+static char *sssctl_config_snippet_path(TALLOC_CTX *ctx, const char *path)
+{
+    char *tmp;
+    const char *snippet_dir;
+    const char delimiter = '/';
+    char *dpos;
+
+    tmp = talloc_strdup(ctx, path);
+    if (!tmp) {
+        return NULL;
+    }
+
+    snippet_dir = strrchr(CONFDB_DEFAULT_CONFIG_DIR, delimiter);
+    if (snippet_dir != NULL) {
+        ++snippet_dir;
+    } else {
+        snippet_dir = "conf.d";
```

Comment:
Not sure if "conf.d" should not be new #define included from src/confdb/confdb.h rather than hardcoded here.

Reply:
Snippet dir is actually created from #define CONFDB_DEFAULT_CONFIG_DIR.
This is fallback when packager breaks the define. It should never happened and I guess that this branch of code is actually removed by optimization if the define is OK.

Reply:
`
41 #define CONFDB_DEFAULT_CFG_FILE_VER 2
42 #define CONFDB_FILE "config.ldb"
43 #define SSSD_CONFIG_FILE SSSD_CONF_DIR"/sssd.conf"
44 #define CONFDB_DEFAULT_CONFIG_DIR SSSD_CONF_DIR"/conf.d"
`
This is part of src/confdb/confdb.h +44
How about declaring new define for "conf.d" there?
CONFDB_DEFAULT_CONFIG_DIR could be concatenated with it and we could use this new define here. This way any time in the future someone will change "conf.d" to something else this will be automatically in sync.

I just not feel safe with keeping two exact strings in two different places in code assuming they will always be the same.

Reply:
OK, I split the definition and I did it also for SSSD_CONFIG_FILE to have some consistency.
The code is now shorter so I like it :)